specifyweb/businessrules/uniqueness_rules.py

```diff
@@ -182,7 +182,7 @@ def create_uniqueness_rule(model_name, discipline, is_database_constraint, field
     UniquenessRuleField = registry.get_model(
         'businessrules', 'UniquenessRuleField') if registry else models.UniquenessRuleField
 
-    created_rule, _ = UniquenessRule.objects.get_or_create(discipline_id=discipline.id if discipline else None,
+    created_rule = UniquenessRule.objects.create(discipline=discipline,
```

Comment:
using `get_or_create` here breaks a couple of tests

Reply:
Fixed the bug. 
The problem here was that when there was a uniqueness rule in a discipline which had the same model_name and is_database_constraint as another uniqueness rule which was already created - even if the rules had different fields/scopes - it would fetch that rule and add the "new rule's" fields/scopes to the already existing scope. 

```json
    "Collection": [
        {
            "rule": [["collectionName"], ["discipline"]],
            "isDatabaseConstraint": false
        },
        {
            "rule": [["code"], ["discipline"]],
            "isDatabaseConstraint": false
        }
    ]
```

So for the above two rules on Collection, the `Collection -> collectionName must be unique in discipline` rule was created first, but then since `Collection -> code must be unique in discipline` had the same model, assigned discipline, and isDatabaseConstraint, it instead altered the `Collection -> collectionName must be unique in discipline` rule to be `Collection -> collectionName and code must be unique to discipline and discipline`.

Reply:
Thank you!